lib/Analysis/Allocation.cpp

```diff
@@ -545,12 +543,13 @@ class AllocationAnalysis {
         auto xRange = bufferRange.lookup(buffer);
         // TODO(Keren): A buffer's size shouldn't be determined here, have to
         // clean it up
-        size_t alignment = buffer->alignment;
-        size_t alignSize = ((size + alignment - 1) / alignment) * alignment;
-        bufferStart[buffer] = alignSize;
-        tripleMap.insert({alignSize + xSize,
-                          Interval{std::max(range.start(), xRange.start()),
-                                   std::min(range.end(), xRange.end())}});
+        size_t offset = size;
+        if (size_t diff = offset % buffer->alignment)
+          offset += buffer->alignment - diff;
```

Comment:
What's the difference between this and the previous calculation?

Reply:
```
        if (size_t diff = newOffset % x->alignment) {
          // fix alignment
          newOffset += x->alignment - diff;
        }
```

I'm fine with the change. It's just that:

1. I'm curious about the differences
2. And maybe you could make this as a function as it's being used multiple times (here and below).

Reply:
It gets the same value, just a different way to compute.

Reply:
I'll make it a function.